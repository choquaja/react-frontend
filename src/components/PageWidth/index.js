import styled, { css } from 'styled-components';
import PropTypes from 'prop-types';

const setWidth = (props) => {
  if (props.fluid) return css`max-width: none;`;
  if (props.small) return css`max-width: 80rem;`;
  return css`max-width: 110rem;`;
};

const PageWidth = styled.div`
  ${setWidth}
  margin: 0 auto;
  height: 100%;
`;

PageWidth.propTypes = {
  fluid: PropTypes.bool,
  small: PropTypes.bool,
};

export default PageWidth;
