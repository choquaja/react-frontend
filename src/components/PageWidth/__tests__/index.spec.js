import React from 'react';
import { shallow } from 'enzyme';
import PageWidth from '../';

describe('PageWidth', () => {
  it('renders without crashing', () => {
    shallow(<PageWidth />);
  });

  it('renders with fluid set true', () => {
    const component = shallow(<PageWidth fluid />);
    expect(component).toMatchSnapshot();
  });

  it('renders with small set true', () => {
    const component = shallow(<PageWidth small />);
    expect(component).toMatchSnapshot();
  });
});
