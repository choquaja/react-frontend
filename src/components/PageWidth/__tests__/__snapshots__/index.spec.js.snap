// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageWidth renders with fluid set true 1`] = `
<div
  className="PageWidth__PageWidth-GUDLM bfwNHj"
/>
`;

exports[`PageWidth renders with small set true 1`] = `
<div
  className="PageWidth__PageWidth-GUDLM crGNoI"
/>
`;
